splice_si_api/src/main/java/com/splicemachine/si/impl/store/IgnoreTxnSupplier.java

```diff
@@ -0,0 +1,112 @@
+
+/*
+ * Copyright (c) 2012 - 2017 Splice Machine, Inc.
+ *
+ * This file is part of Splice Machine.
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.splicemachine.si.impl.store;
+
+import com.splicemachine.access.api.PartitionAdmin;
+import com.splicemachine.access.api.PartitionFactory;
+import com.splicemachine.access.configuration.HBaseConfiguration;
+import com.splicemachine.encoding.MultiFieldDecoder;
+import com.splicemachine.si.api.data.TxnOperationFactory;
+import com.splicemachine.storage.*;
+import com.splicemachine.utils.SpliceLogUtils;
+import org.apache.hadoop.hbase.util.Pair;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Created by jyuan on 10/30/17.
+ */
+public class IgnoreTxnSupplier {
+    private Set<Pair<Long, Long>> cache;
+    private EntryDecoder entryDecoder;
+    final private PartitionFactory partitionFactory;
+    final private TxnOperationFactory txnOperationFactory;
+    private boolean ignoreTxnTableExists;
+    private boolean initialized = false;
+
+    public IgnoreTxnSupplier(PartitionFactory partitionFactory, TxnOperationFactory txnOperationFactory) {
+        this.partitionFactory = partitionFactory;
+        this.txnOperationFactory = txnOperationFactory;
+        cache = new HashSet<>();
+    }
+
+    public boolean shouldIgnore(Long txnId) throws IOException{
+
+        boolean ignore = false;
+        if (!initialized) {
+            init();
```

Comment:
I wouldn't use lazy initialization here, it introduces too many issues for little gain. If you want to use lazy init you have to make sure the init() is properly synchronized, you'd need double checked locking, etc. etc. 

I'd just initialize the cache when we create it, it's not an expensive operation.